### PR TITLE
Re-organize modules in ballerinax/salesforce package

### PR DESCRIPTION
**Pull request by aashikam**
# Description
**Note:**  The improvements required are **known** and are tracked here: https://github.com/ballerina-platform/ballerina-library/issues/6697 and will be a future enhancement. 

**Note:**  A minor version bump. 

Fixes: https://github.com/ballerina-platform/ballerina-library/issues/6691

**Modules**
- [x] bulkv1
- [x] bulkv2 
- [x] apex
- [x] soap
- [x] Deprecate bulkv2 APIs in default package
- [x] Deprecate apex APIs in default package
- [x] Clean up new modules
- [x] Apex API docs
- [x] Bulkv2 API docs

### CODE REVIEW

File: ballerina/modules/apex/client.bal [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/jballerina.java;
+import ballerinax/'client.config;
+import ballerinax/salesforce.utils;
+
+# Ballerina Salesforce connector provides the capability to access Salesforce REST API.
```
Comment (niveathika) [inline ballerina/modules/apex/client.bal:22]: Shouldnt this be apex api?

File: ballerina/modules/apex/client.bal [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    # token or http:OAuth2RefreshTokenGrantConfig if you have Oauth tokens.
+    # Create a Salesforce account and obtain tokens following 
+    # [this guide](https://help.salesforce.com/articleView?id=remoteaccess_authenticate_overview.htm). 
+    #
+    # + salesforceConfig - Salesforce Connector configuration
+    # + return - `sfdc:Error` on failure of initialization or else `()`
+    public isolated function init(ConnectionConfig config) returns error? {
+        http:Client|http:ClientError|error httpClientResult;
+        http:ClientConfiguration httpClientConfig = check config:constructHTTPClientConfig(config);
+        httpClientResult = trap new (config.baseUrl, httpClientConfig);
```
Comment (niveathika) [inline ballerina/modules/apex/client.bal:39]: Is there a bug?Why do we need trap for initalization?

File: ballerina/modules/apex/client.bal [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        if response.statusCode == 200 || response.statusCode == 201 {
+            if response.getContentType() == "" {
+                return;
+            }
+            json responsePayload = check response.getJsonPayload();
+            return check responsePayload.cloneWithType(returnType);
+        } else {
+            json responsePayload = check response.getJsonPayload();
+            return error("Error occurred while executing the apex request. ",
+                httpCode = response.statusCode, details = responsePayload);
```
Comment (niveathika) [inline ballerina/modules/apex/client.bal:94]: Do we need to typecast this? Shouldnt this be handled in the http call itself?

File: ballerina/modules/bulkv2/client.bal [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+
+import ballerina/http;
+import ballerina/io;
+import ballerina/lang.runtime;
+import ballerinax/'client.config;
+import ballerinax/salesforce.utils;
+
+# Ballerina Salesforce connector provides the capability to access Salesforce REST API.
+# This connector lets you to perform operations for SObjects, query using SOQL, search using SOSL, and describe SObjects
+# and organizational data.
```
Comment (niveathika) [inline ballerina/modules/bulkv2/client.bal:25]: Same

File: ballerina/modules/apex/client.bal (hunk 3 above)
Comment (aashikam) [reply ballerina/modules/apex/client.bal:94]: Mentioned in https://github.com/ballerina-platform/ballerina-library/issues/6697

File: ballerina/client.bal [hunk 5]
```diff
@@ -483,6 +483,10 @@ public isolated client class Client {
     # + payload - Payload
     # + returnType - The payload type, which is expected to be returned after data binding
     # + return - `string|int|record{}` type if successful or else `error`
+    # # Deprecated
+    # This function is deprecated due to the introduction of the new submodule salesforce.apex, which supports this API. 
+    # This API will be removed with the 9.0.0 release.
```
Comment (NipunaRanasinghe) [inline ballerina/client.bal:488]:
```suggestion
    # # Deprecated
    # This function is deprecated due to the introduction of the new submodule `salesforce.apex`, which supports the same functionality. 
    # This API will be removed with the 9.0.0 release.
```

Comment (NipunaRanasinghe) [reply ballerina/client.bal:488]: @aashikam shall we update the similar places accordingly?

File: ballerina/modules/apex/Module.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
+## Overview
+
+Salesforce Apex REST API enables you to expose your Apex classes and methods as RESTful web services. This module provides operations for executing custom Apex REST endpoints, allowing you to perform various HTTP operations on these endpoints and handle responses accordingly.
```
Comment (NipunaRanasinghe) [inline ballerina/modules/apex/Module.md:3]: Wouldn't it be better to mention the Apex API version we currently support?

File: ballerina/modules/apex/Module.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+apex:Client apexClient = check new (sfConfig);
+```
+
+### Step 3: Invoke connector operation
+
+1. Now you can use the operations available within the connector. Note that they are in the form of remote operations.
+Following is an example of how to execute a custom Apex REST endpoint using the connector.
+
+```ballerina
+public function main() returns error? {
```
Comment (niveathika) [inline ballerina/modules/apex/Module.md:81]: Shall we format the example scripts

File: ballerina/modules/apex/Module.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    string|error caseId = apexClient->apexRestExecute("Cases", "POST",
```
Comment (ayeshLK) [inline ballerina/modules/apex/Module.md:82]: Do we need to handle the error case in this code-snippet ? I think we can only consider the happy path and structure the code like this;
```ballerina
    string caseId = check apexClient->apexRestExecute("Cases", "POST", ...);
    // handle the happy flow here
```

File: ballerina/modules/apex/client.bal [hunk 9]
```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (ayeshLK) [inline ballerina/modules/apex/client.bal:1]: Check for other places as well.
```suggestion
// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.org).
```

File: ballerina/modules/apex/client.bal [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+    private map<string> sfLocators = {};
+
+    # Initializes the Salesforce Apex Client. During initialization, you can pass either `http:BearerTokenConfig`
+    # if you have a bearer token or `http:OAuth2RefreshTokenGrantConfig` if you have OAuth tokens.
+    # Create a Salesforce account and obtain tokens following 
+    # [this guide](https://help.salesforce.com/articleView?id=remoteaccess_authenticate_overview.htm).
+    #
+    # + config - Salesforce connector configuration
+    # + return - `error` on failure of initialization or else `()`
+    public isolated function init(ConnectionConfig config) returns error? {
```
Comment (ayeshLK) [inline ballerina/modules/apex/client.bal:36]: Can't we use included record params here ?
```ballerina
    public isolated function init(*ConnectionConfig config) returns error? {
        // implement the init
    }
```

File: ballerina/modules/apex/client.bal [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+        http:Client|http:ClientError|error httpClientResult;
+        http:ClientConfiguration httpClientConfig = check config:constructHTTPClientConfig(config);
+        httpClientResult = trap new (config.baseUrl, httpClientConfig);
```
Comment (ayeshLK) [inline ballerina/modules/apex/client.bal:39]: IMO we can simplify the code like this.
```suggestion
        http:ClientConfiguration httpClientConfig = check config:constructHTTPClientConfig(config);
        http:Client|error httpClientResult = new (config.baseUrl, httpClientConfig);
```

File: ballerina/modules/apex/tests/test.bal [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+function testApex() returns error? {
+    log:printInfo("baseClient -> executeApex()");
+    string|error caseId = baseClient->apexRestExecute("Cases", "POST", 
+        {"subject" : "Bigfoot Sighting9!",
+            "status" : "New",
+            "origin" : "Phone",
+            "priority" : "Low"});
+    if caseId is error {
+        test:assertFail(msg = caseId.message());
+    }
```
Comment (ayeshLK) [inline ballerina/modules/apex/tests/test.bal:54]:
```suggestion
    string caseId = check baseClient->apexRestExecute("Cases", "POST", 
        {"subject" : "Bigfoot Sighting9!",
            "status" : "New",
            "origin" : "Phone",
            "priority" : "Low"});
```

File: ballerina/modules/apex/tests/test.bal [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+    runtime:sleep(5);
+    record{}|error case = baseClient->apexRestExecute(string `Cases/${caseId}`, "GET", {});
+    if case is error {
+        test:assertFail(msg = case.message());
+    }
```
Comment (ayeshLK) [inline ballerina/modules/apex/tests/test.bal:59]:
```suggestion
    record{} case = check baseClient->apexRestExecute(string `Cases/${caseId}`, "GET", {});
```

File: ballerina/modules/apex/tests/test.bal [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+    runtime:sleep(5);
+    error? deleteResponse = baseClient->apexRestExecute(string `Cases/${caseId}`, "DELETE", {});
+    if deleteResponse is error {
+        test:assertFail(msg = deleteResponse.message());
+    }
```
Comment (ayeshLK) [inline ballerina/modules/apex/tests/test.bal:64]: Can't we follow this approach ?
```suggestion
    test:assertFalse(deleteResponse is error, "Error occurred wile deleting");
```

File: ballerina/modules/bulkv2/Module.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+Following is an example of how to create a bulk job using the connector.
+
+```ballerina
+bulkv2:BulkCreatePayload payload = {
+    'object : "Contact",
+    contentType : "CSV",
+    operation : "insert",
+    lineEnding : "LF"
+};
+error|bulkv2:BulkJob insertJob = baseClient->createIngestJob(payload);
```
Comment (ayeshLK) [inline ballerina/modules/bulkv2/Module.md:82]:
```suggestion
bulkv2:BulkJob insertJob = check baseClient->createIngestJob(payload);
```

File: ballerina/modules/bulkv2/client.bal [hunk 16]
```diff
@@ -0,0 +1,295 @@
...
+    private map<string> sfLocators = {};
+
+    # Initializes the Bulk V2 client. During initialization you can pass either http:BearerTokenConfig if you have a bearer
+    # token or http:OAuth2RefreshTokenGrantConfig if you have Oauth tokens.
+    # Create a Salesforce account and obtain tokens following 
+    # [this guide](https://help.salesforce.com/articleView?id=remoteaccess_authenticate_overview.htm). 
+    #
+    # + salesforceConfig - Salesforce Connector configuration
+    # + return - `sfdc:Error` on failure of initialization or else `()`
+    public isolated function init(ConnectionConfig config) returns error? {
```
Comment (ayeshLK) [inline ballerina/modules/bulkv2/client.bal:38]:
```suggestion
    public isolated function init(*ConnectionConfig config) returns error? {
```

**Comment by aashikam** [comment]
@ayeshLK and all Just FYI before you guys review all the code. This is just moving around existing code. We need to move everything for update 10 as this a minor bump. We have used the same client in the default package. https://github.com/ballerina-platform/module-ballerinax-salesforce/blob/e9bc4ee3f5be51532da97bdd20f9ed958d689d11/ballerina/client.bal#L39. Tests are also the existing tests, which are just being moved from default package to modules. 

We have no intention in improving the existing code quality in this effort, this will just be a move. All the errors and improvements required are known and tracked here https://github.com/ballerina-platform/ballerina-library/issues/6697 and I will take a look at all this at once in a separate PR, as there is a **lot**. Please keep reviewing I will add some of the changes to this issue.

File: ballerina/modules/apex/Module.md (hunk 6 above)
Comment (aashikam) [reply ballerina/modules/apex/Module.md:3]: it is here: the same functionality., Added in here also.

File: ballerina/modules/apex/client.bal (hunk 9 above)
Comment (aashikam) [reply ballerina/modules/apex/client.bal:1]: Hey why is this wrong, I checked other modules also

Comment (aashikam) [reply ballerina/modules/apex/client.bal:1]: https://github.com/niveathika/module-ballerinax-sap.s4hana.procurement/blob/9677c6118942096e6aa12c4f2068350fa873f4de/docs/sanitation/sanitations.bal

File: ballerina/modules/apex/client.bal (hunk 10 above)
Comment (aashikam) [reply ballerina/modules/apex/client.bal:36]: We will add this as a future enhancement.

File: ballerina/modules/apex/client.bal (hunk 9 above)
Comment (ayeshLK) [reply ballerina/modules/apex/client.bal:1]: Pls check this email [1]

[1] - `Re: License Header for WSO2 code`

**Review by niveathika — DISMISSED** [review]
LGTM. @aashikam Lets update the changelog.

**Review by ayeshLK — COMMENTED** [review]
LGTM

File: ballerina/modules/apex/Module.md [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+
+2. Go to Setup --> Apps --> App Manager 
+
+   <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-sfdc/master/docs/setup/resources/side-panel.png alt="Setup Side Panel" width="40%" style="border:1px solid #000000">
+
+3. Create a New Connected App.
+
+   <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-sfdc/master/docs/setup/resources/create-connected-apps.png alt="Create Connected Apps" width="50%" style="border:1px solid #000000">
+
+    - Here we will be using https://test.salesforce.com as we are using sandbox environment. Users can use https://login.salesforce.com for normal usage.
```
Comment (RDPerera) [inline ballerina/modules/apex/Module.md:19]:
```suggestion
    - Here we will be using https://test.salesforce.com as we are using a sandbox environment. Users can use https://login.salesforce.com for normal usage.
```

**Review by niveathika — APPROVED** [review]

File: ballerina/modules/apex/Module.md [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+
+    <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-sfdc/master/docs/setup/resources/create_connected%20_app.png alt="Create Connected Apps" width="100%" style="border:1px solid #000000">
+
+4. After the creation user can get consumer key and secret through clicking on the `Manage Consumer Details` button.
```
Comment (RDPerera) [inline ballerina/modules/apex/Module.md:23]:
```suggestion
4. After the creation, the user can get the consumer key and secret by clicking on the `Manage Consumer Details` button.
```

File: ballerina/modules/apex/Module.md [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+   <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-sfdc/master/docs/setup/resources/crdentials.png alt="Consumer Secrets" width="100%" style="border:1px solid #000000">
+
+5. The next step is to get the token.
+
+    - Log in to Salesforce in your preferred browser and enter the following URL:
+  `https://<YOUR_INSTANCE>.salesforce.com/services/oauth2/authorize?response_type=code&client_id=<CONSUMER_KEY>&redirect_uri=<REDIRECT_URL>`
+   - Allow access if an alert pops up, and the browser will be redirected to a URL like the following:
+  `https://login.salesforce.com/?code=<ENCODED_CODE>`
+  
+   - The code can be obtained after decoding the encoded code
```
Comment (RDPerera) [inline ballerina/modules/apex/Module.md:34]:
```suggestion
   - The code can be obtained after decoding the encoded code.
```

File: ballerina/modules/apex/Module.md [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+        {
+        "subject": "Item Fault!",
+        "status": "New",
+        "priority": "High"
+    });
+    return;
+}
+```
+
+2. Use `bal run` command to compile and run the Ballerina program.
```
Comment (RDPerera) [inline ballerina/modules/apex/Module.md:88]:
```suggestion
2. Use the `bal run` command to compile and run the Ballerina program. 
```

**Pull request merged**